### DISCUSSION TITLE
web3 modal and calling with contract

### DISCUSSION THREAD

**Discussion by Sriharsh1103** [WalletConnect/walletconnect-monorepo#2906] [Developer Support] [answered]
i need a demo on in react connectwalletv2 web3 modal functional calling with contract and how can i customize web3 modal according to our project need ? if any plz share link and or code soo we could implement it

**Comment by ignaciosantise** [comment, marked as answer]
hey @Sriharsh1103 as you can read here https://docs.walletconnect.com/2.0/web/web3modal/react/wagmi/installation#use-wagmi-hooks you can use wagmi hooks to interact with Contracts. 

Here you can find an example to read a contract: https://wagmi.sh/react/hooks/useContractRead

**Reply by Sriharsh1103** [reply]
their is problem in activeconnector .. which is used for account switching and chain switching from mobile sometimes it is working sometimes it is not working . what's the reason behind it. is code is not stable yet or i am making any mistake... it urgent...
thank for above response hope get solution regarding this soon .

**Reply by Sriharsh1103** [reply]
and when i switch chain id it is not triggering activeconnector i didn't get pop-up for network switch

**Reply by Sriharsh1103** [reply]
@Ellinka779 @ignaciosantise >>> reply for above soon thanks

**Comment by Sriharsh1103** [comment]
should we use the below code snippet where we are passing custom RPC's?? or we don't need to pass our RPC in wagmi..? 

import { chain, configureChains } from 'wagmi'
import { jsonRpcProvider } from 'wagmi/providers/jsonRpc'

const { chains, provider } = configureChains(
 [chain.testnet, chain.caldera],
 [
     jsonRpcProvider({
         rpc: (chain) => ({
             http: `RPC URL`, // Insert RPC URL here
             }),
         }),
     ],
)
 
and any active developer response if possible

**Comment by ignaciosantise** [comment]
@Sriharsh1103 can you provide a reproducible example? In a public repository or in codesanbox.
